### PR TITLE
Fix events for sdk resolvers load

### LINKED ISSUE #9679: List of SDK resolvers should be immutable in a given process
It's not the primary cause of the wasted time in #9678, but it's surprising that MSBuild _ever_ attempts to rediscover and reload SDK resolvers. Since the manifests are in one of the treat-as-immutable-for-the-process-lifetime cones, we should identify resolvers at most once in a process lifetime.

### PR DESCRIPTION
**Pull request by AR-May**
Fixes #9679

### Context
It seems like events in sdk resolver service are misleading. We emit some events even when resolvers were taken from process-wide cache and not actually loaded.

### Changes Made
Moved the events emission to SdkResolverLoader, so that the events concerning loading are not emitted when a cached value was used.

### CODE REVIEW

**Review by f-alizada — COMMENTED** [review]
Thank youf ror the PR! Overall looks good, however will it be possible to add testing for the changed code?

File: src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs [hunk 1]
```diff
@@ -36,15 +37,20 @@ internal class SdkResolverLoader
 
         internal virtual IReadOnlyList<SdkResolver> GetDefaultResolvers()
         {
+
+            MSBuildEventSource.Log.SdkResolverLoadResolversStart();
             var resolvers = !string.Equals(IncludeDefaultResolver, "false", StringComparison.OrdinalIgnoreCase) ?
                 new List<SdkResolver> { new DefaultSdkResolver() }
                 : new List<SdkResolver>();
 
+            MSBuildEventSource.Log.SdkResolverLoadResolversStop(string.Empty, resolvers.Count);
```
Comment (ladipro) [inline src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs:46]: I don't think this is worth measuring. It's just an instantiation of a couple of objects.

File: src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs [hunk 2]
```diff
@@ -62,13 +68,18 @@ internal virtual IReadOnlyList<SdkResolver> LoadAllResolvers(ElementLocation loc
                 LoadResolvers(potentialResolver, location, resolvers);
             }
 
+            MSBuildEventSource.Log.SdkResolverLoadAllResolversStop(resolvers.Count);
```
Comment (ladipro) [inline src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs:71]: nit: I wonder if this shouldn't be moved to a `finally` block. Same for the other Stop calls in this file.

File: src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs (hunk 1 above)
Comment (AR-May) [reply src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs:46]: It also it reports the number of resolvers loaded and without these events the log would not report the correct number of loaded resolvers. But we are probably not super interested in that, so theoretically we can drop them.

**Comment by AR-May** [comment]
> Thank youf ror the PR! Overall looks good, however will it be possible to add testing for the changed code?

Only logging is changed, I believe that we do not cover it with tests.

File: src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs (hunk 1 above)
Comment (ladipro) [reply src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs:46]: Yes, we're not really _loading_ any resolvers here so I think we wouldn't miss anything without this event.

**Comment by ladipro** [comment]
We currently don't test our `EventSource` event logging but it would be nice if we did. May be worth filing an issue for it.

File: src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs (hunk 2 above)
Comment (AR-May) [reply src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs]: I agree, that's a good pattern to have to make sure that the stop event was sent. Fixed.

**Comment by AR-May** [comment]
> We currently don't test our `EventSource` event logging but it would be nice if we did. May be worth filing an issue for it.

I agree, let's file an issue for that and discuss this option.

File: src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs (hunk 1 above)
Comment (AR-May) [reply src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs]: Agreed, removed them.

File: src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs [hunk 3]
```diff
@@ -39,36 +40,52 @@ internal virtual IReadOnlyList<SdkResolver> GetDefaultResolvers()
...
             }
 
             return resolvers.OrderBy(t => t.Priority).ToList();
         }
 
         internal virtual IReadOnlyList<SdkResolverManifest> GetResolversManifests(ElementLocation location)
         {
-            return FindPotentialSdkResolversManifests(
+            MSBuildEventSource.Log.SdkResolverFindResolversManifestsStart();
+            IReadOnlyList<SdkResolverManifest> allResolversManifests = null;
```
Comment (ladipro) [inline src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs:78]: nit: You could still keep the `return FindPotentialSdkResolversManifests(...` inside the `try` block. No need to introduce a local variable.

File: src/Framework/MSBuildEventSource.cs [hunk 4]
```diff
@@ -476,13 +476,13 @@ public void WriteLinesToFileUpToDateStop(string fileItemSpec, bool wasUpToDate)
...
         [Event(62, Keywords = Keywords.All)]
-        public void SdkResolverServiceInitializeStart()
+        public void SdkResolverLoadAllResolversStart()
         {
             WriteEvent(62);
         }
 
         [Event(63, Keywords = Keywords.All)]
-        public void SdkResolverServiceInitializeStop(int resolverCount)
+        public void SdkResolverLoadAllResolversStop(int resolverCount)
```
Comment (ladipro) [inline src/Framework/MSBuildEventSource.cs:485]: @jeffkl, I believe that you added some of these events. Can you please confirm that renaming is ok, i.e. nothing on your side depends on the names. Thank you!

**Review by ladipro — APPROVED** [review]

File: src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs (hunk 3 above)
Comment (AR-May) [reply src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs:78]: I need access to resolvers count for the Stop event in the `finally` block. So, I think that we need the local variable here.

Comment (ladipro) [reply src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs:78]: Of course! Please ignore me.

**Review by rokonec — APPROVED** [review]

File: src/Framework/MSBuildEventSource.cs (hunk 4 above)
Comment (jeffkl) [reply src/Framework/MSBuildEventSource.cs:485]: I am not personally aware of any dependency on the names of these events

**Review by jeffkl — APPROVED** [review]

File: src/Framework/MSBuildEventSource.cs (hunk 4 above)
Comment (rainersigwald) [reply src/Framework/MSBuildEventSource.cs:485]: My read of https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventsource-instrumentation#best-practices rule 7 is that renaming should imply giving these new numbers, but it's not that clear to me.

**Pull request merged**